### DISCUSSION TITLE
BarDropdown and Dropdown inconsistency

### DISCUSSION THREAD

**Discussion by johndcollins** [Megabit/Blazorise#3309] [Q&A] [answered]
I enjoy this project, but I have a few issues.

Why is the BarDropdown and the Dropdown so different? Why can the interface be similar?
I want to place a menu at the top right of my header, but I have to go through hoops to try and get the BarDropdown menu to show to the left where this work with a normal Dropdown. Also setting properties is different on BarDropdown and Dropdown like the RightAligned property.

Also, why is there no option to expand a submenu on hover instead of click?

**Comment by stsrki** [comment]
Did you try the `RightAligned` parameter to open the menu to the left? eg.

```cshtml
<BarDropdown>
  <BarDropdownToggle>Toggle</BarDropdownToggle>
  <BarDropdownMenu RightAligned="true">
      ...
  </BarDropdownMenu>
</BarDropdown>
```

Can you explain what exactly you find different between BarDropdown and Dropdown? They belong to a separate components but their usage structure is still similar, if not the same.

---

> Also, why is there no option to expand a submenu on hover instead of click?

This is something we can discuss for future versions.

**Comment by johndcollins** [comment]
On the regular Dropdown the RightAligned is on the DropDown Tag where on the BarDropDown it's on the BarDropDownMenu Tag

The Documentation does say the RightAligned property is on both the  DropDown Tag and the DropDownMenu Tag for the regular DropDown, but that is not the case.

The DropDown also has a property called Direction so you can set the direction of the menu, but the BarDropDownMenu Tag doesn't have this option.

I did use the RightAligned on the BarDropDownMenu and it does align it, but I'm missing the Direction property.

**Comment by stsrki** [comment, marked as answer]
Hmm, yes, you're right. We need to adjust the API to be the same on both components. I will try to make it for the v1.0 release. Thanks for taking the time to point us in the right direction.
